### PR TITLE
Improved conversion of MapInfo line symbols

### PR DESCRIPTION
**Pull request by nyalldawson**
Many MapInfo line symbols don't translate well into OGR styles, so
in order to improve the quality of the conversion this commit introduces
a new QgsMapInfoSymbolConverter class which contains utility functions
for converting MapInfo symbols by ID to their QGIS equivalent.

Implemented for line symbols only for now.

### CODE REVIEW

File: src/core/qgsogrutils.cpp [hunk 1]
```diff
@@ -1256,6 +1257,18 @@ std::unique_ptr<QgsSymbol> QgsOgrUtils::symbolFromStyleString( const QString &st
...
 
+    // if the pen is a mapinfo pen, use dedicated converter for more accurate results
+    const thread_local QRegularExpression sMapInfoId = QRegularExpression( QStringLiteral( "mapinfo-pen-(\\d+)" ) );
+    const QRegularExpressionMatch match = sMapInfoId.match( string );
+    if ( match.hasMatch() )
+    {
+      const int penId = match.captured( 1 ).toInt();
+      QgsMapInfoSymbolConversionContext context;
+      std::unique_ptr<QgsSymbol> res( QgsMapInfoSymbolConverter::convertLineSymbol( penId, context, color, lineWidth, lineWidthUnit ) );
+      if ( res )
```
Comment (pblottiere) [inline src/core/qgsogrutils.cpp:1268]: When can we expect the `res` unique_ptr to be invalid in this situation?

File: src/core/symbology/qgsmapinfosymbolconverter.cpp [hunk 2]
```diff
@@ -0,0 +1,1096 @@
...
+      symbol->appendSymbolLayer( hash2.release() );
+    }
+    else if ( identifier == 118 )
+    {
+      qgis::down_cast< QgsSimpleLineSymbolLayer * >( symbol->symbolLayer( 0 ) )->setColor( QColor( 0, 0, 0 ) );
+      qgis::down_cast< QgsSimpleLineSymbolLayer * >( symbol->symbolLayer( 0 ) )->setWidth( 0 );
+      symbol->symbolLayer( 0 )->setLocked( true );
+
+      std::unique_ptr<QgsSimpleLineSymbolLayer > secondRail( qgis::down_cast< QgsSimpleLineSymbolLayer * >( symbol->symbolLayer( 0 ) )->clone() );
+      double offset = 2 * size;
```
Comment (pblottiere) [inline src/core/symbology/qgsmapinfosymbolconverter.cpp:464]:
```suggestion
      const double offset = 2 * size;
```

File: src/core/qgsogrutils.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/qgsogrutils.cpp:1268]: There's some symbols we can't reproduce in qgis (zig zag lines), so there's a chance the converter will return nullptr

Comment (pblottiere) [reply src/core/qgsogrutils.cpp:1268]: (thanks for the explanation, I didn't see the *zig-zaggy patterns* case)

**Review by pblottiere — APPROVED** [review]

**Comment by roya0045** [comment]
I feel like a reference screenshot would be nice to have.

**Pull request merged**